pkg/pillar/cmd/zedrouter/pbr.go

```diff
@@ -281,15 +281,15 @@ func AddOverlayRuleAndRoute(bridgeName string, iifIndex int,
 	return nil
 }
 
-// AddFwMarkRuleToDummy : Create a ip rule that sends packets marked with given mark
+// AddFwMarkRuleToDummy : Create an ip rule that sends packets marked by a Drop ACE
 // out of interface with given index.
-func AddFwMarkRuleToDummy(fwmark uint32, iifIndex int) error {
+func AddFwMarkRuleToDummy(iifIndex int) error {
 
 	r := netlink.NewRule()
 	myTable := baseTableIndex + iifIndex
 	r.Table = myTable
-	r.Mark = int(fwmark)
-	r.Mask = 0x00ffffff
+	r.Mark = aceDropAction
```

Comment:
What about the case where there are no rules with DROP action and the packet did not match any of the configured ACL rules?
If my guess is right, they get marked with 0x7fffff
Is there a separate ip rule for that?

Reply:
The ACE ID and the action to take are now *separate bits*:
- the lowest 23 bits are the ACE ID (for the default drop it is 0x7fffff indeed)
- the next bit is the action to take (1 -> drop, 0 -> allow)

Therefore there is only one IP rule that matches packets if the 24th bit of connmark is set to 1 (drop action). This single rule works for both explicit DROP ACE as well as for the implicit default-drop ACE.